### ISSUE TITLE
level filtering not working

### ISSUE THREAD

**Issue by maxigit** [flycheck/flycheck#955]
I'm setting the level filtering in flycheck using the f key in error list and nothing happened. When I set it to error I expect all the warning to disappear but they are still there. Also , when I go to the next error, warning are not skipped. (I have the problem with haskell code).

**Comment by cpitclaudel** [comment]
Thanks for this report. Can you provide a code sample demonstrating this issue (Haskell would be fine)? I cannot reproduce it on my machine.

I'm not sure I understand your report completely, but this could also be due to a miscommunication about what this feature does. At the moment, filtering only affects the error list; it does not hide error annotations (squiggles) from the source buffer, nor does it change the behaviour of navigation commands.

**Comment by maxigit** [comment]
I'm not bothered about the squiggles, but more the navigation. Setting `flyckeck-navigation-minimum-level` or `flycheck-error-list-minimum-level` doesn't seem to have any effect, whatsoever.

I have the problem with every files (haskell or not) but I'm using spacemacs. You could try with this


```
-- should issue an error, `f` lacks an accompanying binding
f :: Int -> Int
-- should issue an info, redundant bracket
f' x = (x + 1)
```

using the `haskell-ghc` checker.

**Comment by lunaryorn** [comment]
How did you set it, and how do you navigate?  It'd be great if you could provide a [reproducible `emacs -Q` recipe](http://www.lunaryorn.com/2015/11/29/reproduce-bugs-in-emacs-Q.html) that illustrates the behaviour you get, as opposed to the behaviour you'd like to see.

**Comment by maxigit** [comment]
I navigate using spacemacs shortcut : SPC e n. I set it either using the f
key on the error list or the `flycheck-set-error-list-filter-level` or
customize-variable`.
I'll try to reproducte it using emacs -Q, but that things difficult as I
need flycheck, haskell and possibly spacemacs to reproduce it.

**Comment by lunaryorn** [comment]
@maxigit You won't need Spacemacs to reproduce it.  Or in other words, if you need Spacemacs to reproduce it it's a Spacemacs issue, not a Flycheck problem.

Please do try to reproduce the behaviour without Spacemacs.  We can't give support for Spacemacs, given the amount of things it changes.

**Comment by maxigit** [comment]
It's a Spacemacs problem. The severity is not set when the error are defined.

**Comment by lunaryorn** [comment]
Please report the issue to Spacemacs in this case.  Thank you for taking the time to debug the issue, I hope that the problem in Spacemacs gets fixed fast.

Thank you.

**Comment by maxigit** [comment]
I sent a pull request to fix it.

**Comment by syl20bnr** [comment]
Thank you :+1:
